### ISSUE TITLE
missing sea ice file

### ISSUE THREAD

**Issue by tangq** [E3SM-Project/E3SM#4848]
When attempting to use the current (3/20/2022) master to fix the OpenMPI issue (https://github.com/E3SM-Project/E3SM/pull/4823) for the [NA RRM AMIP run](https://acme-climate.atlassian.net/wiki/spaces/EWCG/pages/3302982236/v2.NARRM.amip) as suggested by @ambrad, @rljacob , I got the missing mpassi file error on chrysalis. Does anyone know where this file is located?

```
Model mpassi missing file graph128 = '/lcrc/group/e3sm/data/inputdata/ice/mpas-cice/WC14to60E2r3/mpas-seaice.graph.info.200714.part.128'
Trying to download file: 'ice/mpas-cice/WC14to60E2r3/mpas-seaice.graph.info.200714.part.128' to path '/lcrc/group/e3sm/data/inputdata/ice/mpas-cice/WC14to60E2r3/mpas-seaice.graph.info.200714.part.128' using WGET protocol.
wget failed with output:  and errput --2022-03-20 14:31:02--  https://web.lcrc.anl.gov/public/e3sm/inputdata/ice/mpas-cice/WC14to60E2r3/mpas-seaice.graph.info.200714.part.128
Resolving web.lcrc.anl.gov (web.lcrc.anl.gov)... 140.221.70.30
Connecting to web.lcrc.anl.gov (web.lcrc.anl.gov)|140.221.70.30|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2022-03-20 14:31:02 ERROR 404: Not Found.
```

**Comment by rljacob** [comment]
Make sure you use the new maint-2.0 branch for any v2.0 runs.  It has that fix.

Confirmed that file isn't on the server but if you're continuing a run, why would the number of sea ice tasks be one you haven't used before?

**Comment by ambrad** [comment]
I agree with Rob. Also, something is amiss if the runs wants to use just 128 ranks for sea ice, as indicated by ".part.128".

**Comment by tangq** [comment]
I did a clean test with the maint-2.0 branch (hash: `fe54e14a16d6588135a2890daebcd5b47f5ded29`) and it still complains about missing `mpas-seaice.graph.info.200714.part.128`. This suggests that the maint-2.0 branch is broken at least for running NA RRM AMIP runs.

I noticed that there is an extra `mpas-seaice` directory on the same level as `build`.

The test is at chrysalis: /lcrc/group/e3sm/ac.qtang/E3SMv2/test.v2.NARRM.amip_0201

**Comment by rljacob** [comment]
Do you still have the old case directory that worked?

**Comment by tangq** [comment]
Yes, the old, working case is at /lcrc/group/e3sm/ac.qtang/E3SMv2/v2.NARRM.amip_0201/fba8cd0a29

**Comment by ambrad** [comment]
Re: my comment above re: ".part.128", when I look at the directory Qi links, I see that tests/S_2x5_ndays/case_scripts/env_mach_pes.xml is indeed requesting 128 tasks, in fact for all components. This is not consistent with the S setting for the NA RRM compset. 128 tasks for all components is the default for Chrysalis when the grid and compset don't match anything other than "any". So I think the primary question is why cime isn't recognizing the grid and compset and getting the right PE layout from config_pesall.xml,

**Comment by rljacob** [comment]
Also noticed:  this is an F-case:  --compset F20TR --res northamericax4v1pg2_WC14to60E2r3.  That's not in our test suite.  The coupled version is (--compset WCYCL1850 --res northamericax4v1pg2_WC14to60E2r3).  So this may have been broken for a while. 

If a new pelayout was added to config_pesall.xml that is related (same compset or machine or res), that could confuse the logic that picks the layout.

**Comment by ambrad** [comment]
@tangq is this the first time you've tried to run this as "hybrid"? Perhaps "hybrid" is broken.

**Comment by rljacob** [comment]
Looking at your old case directory, /lcrc/group/e3sm/ac.qtang/E3SMv2/v2.NARRM.amip_0201/fba8cd0a29/case_scripts, it was using 3840 tasks for all components.

**Comment by tangq** [comment]
I think I know why - It's because only the M and L PE layouts are available for NA RRM. I changed the test to M_1x10_ndays and it works! We need a better warning message when the default layout is not available.

Also, the NA RRM F and coupled cases should be tested routinely - They are parts of the official E3SMv2 release.

**Comment by rljacob** [comment]
That's good.  But that case should default to a workable layout when no pelayout is specified.  I'll change the title of this issue and assign it.

**Comment by amametjanov** [comment]
The default pelayout (when `--pecount *` is omitted in `create_newcase`) is `pesize="any"` or `pesize="M"`. Medium and Large RRM-AMIP  pelayouts have been on master since Feb 9 (also on `maint-2.0`) with PR E3SM-Project/E3SM#4783.
If a `create_newcase` requests a non-existent pelayout (`--pecount S` above), then a "next-best" match is attempted: here `mach name` matched with default pelayout for Chrysalis of 128x2 tasks x threads.

To clarify, do we need a Small RRM AMIP pelayout ? Current default is 30 nodes.

**Comment by tangq** [comment]
I don't see a need for the Small RRM AMIP layout. The M and L layouts use (I believe) 30 and 60 nodes layouts, which should be sufficient.

Is there a way to check which PE layouts are available for a certain compset and resolution (e.g, AMIP NA RRM)? If so, users can avoid asking for a non-existent layout.

**Comment by rljacob** [comment]
@amametjanov did you confirm that pesize=M is picked up when a layout is not specified?   Qi's testing indicated that a 128-task layout was what was chosen.   It would be find to just make sure that "M" is the default.

**Comment by rljacob** [comment]
On the other hand if 128 tasks is the default on Chrysalis that we should have a 128 task graph for all MPAS components.

**Comment by tangq** [comment]
> @amametjanov did you confirm that pesize=M is picked up when a layout is not specified? Qi's testing indicated that a 128-task layout was what was chosen. It would be find to just make sure that "M" is the default.

In my previous run script, it specified to use the "S" layout.

**Comment by jonbob** [comment]
@rljacob - we can make the graph files, but some of the grids are too large for the code to run on 128 tasks

**Comment by amametjanov** [comment]
> Is there a way to check which PE layouts are available for a certain compset and resolution (e.g, AMIP NA RRM)?

`cime/scripts/query_config` does not appear to provide that option. Browsing through comments in `cime_config/allactive/config_pesall.xml` shows configured pelayouts, which are also summarized in https://acme-climate.atlassian.net/wiki/spaces/DOC/pages/1008074952/E3SM+v1+SYPD+matrix .

> did you confirm that pesize=M is picked up when a layout is not specified?

yes.

> On the other hand if 128 tasks is the default on Chrysalis that we should have a 128 task graph for all MPAS components.

128x2 is the generic default for any grid + any compset.
I added the missing `.part.128` graph file for mpas-seaice `WC14to60E2r3` grid.
RRM AMIP (`SMS_PS.northamericax4v1pg2_WC14to60E2r3.F2010.chrysalis_intel.allactive-wcprodrrm`) runs 128x2 on 4 nodes at 0.821 sypd.

A default for this grid (`SMS.northamericax4v1pg2_WC14to60E2r3.F2010.chrysalis_intel.allactive-wcprodrrm`) runs on 30 nodes at 6.3 sypd.
